server/src/main/java/org/elasticsearch/search/dfs/DfsSearchResult.java

```diff
@@ -168,6 +168,23 @@ public  static void writeSingleTermStats(StreamOutput out, TermStatistics termSt
                 docCount = subOne(in.readVLong());
                 sumTotalTermFreq = subOne(in.readVLong());
                 sumDocFreq = subOne(in.readVLong());
+                if (sumTotalTermFreq == -1L) {
+                    // fallback used by Lucene in ES 6 (LUCENE-8007)
+                    sumTotalTermFreq = sumDocFreq;
+                }
+                if (maxDoc == 0L) {
+                    // fallback used by Lucene in ES 6 (LUCENE-8007)
+                    maxDoc = docCount;
+                }
```

Comment:
Lucene seems to be doing it the other way arround: `if (docCount == -1) { docCount = maxDoc; }` reading https://github.com/apache/lucene-solr/blob/releases/lucene-solr/7.7.3/lucene/core/src/java/org/apache/lucene/search/similarities/BM25Similarity.java#L99.

Reply:
Makes sense. I'm mainly wondering how to exercise this logic (I can't see a way for this to happen with the codecs that we offer).

Reply:
Right, the default codec would always return a non-negative docCount. Maybe one approach would be to have this case only covered through a unit test that emulates 6.x serialization logic.

Reply:
I've added a unit test in 5abe7d351d9. Not sure it adds much though.